components/channel_layout/channel_controller.tsx

```diff
@@ -1,77 +1,79 @@
-    componentDidMount() {
+const BODY_CLASS_FOR_CHANNEL = ['app__body', 'channel-view'];
+
+export default function ChannelController({shouldShowAppBar, fetchingChannels}: Props) {
+    useEffect(() => {
+        const isMsBrowser = isInternetExplorer() || isEdge();
         const platform = window.navigator.platform;
+        document.body.classList.add(...getClassnamesForBody(platform, isMsBrowser));
 
-        document.body.classList.add('app__body', 'channel-view');
+        return () => {
+            document.body.classList.remove(...BODY_CLASS_FOR_CHANNEL);
+        };
+    }, []);
 
-        // IE Detection
-        if (UserAgent.isInternetExplorer() || UserAgent.isEdge()) {
-            document.body.classList.add('browser--ie');
-        }
+    return (
+        <div
+            id='channel_view'
+            className='channel-view'
+        >
+            <AnnouncementBarController/>
+            <SystemNotice/>
+            <FaviconTitleHandler/>
+            <ProductNoticesModal/>
+            <div className={classNames('container-fluid channel-view-inner', {'app-bar-enabled': shouldShowAppBar})}>
+                <SidebarRight/>
+                <SidebarRightMenu/>
+                <Sidebar/>
+                {fetchingChannels ? <LoadingScreen/> : <CenterChannel/>}
+                <Pluggable pluggableName='Root'/>
+                <ResetStatusModal/>
+            </div>
+            <AppBar/>
+        </div>
+    );
+}
 
-        // OS Detection
-        if (platform === 'Win32' || platform === 'Win64') {
-            document.body.classList.add('os--windows');
-        } else if (platform === 'MacIntel' || platform === 'MacPPC') {
-            document.body.classList.add('os--mac');
-        }
-    }
+export function getClassnamesForBody(platform: Window['navigator']['platform'], isMsBrowser = false) {
+    const bodyClass = [...BODY_CLASS_FOR_CHANNEL];
 
-    componentWillUnmount() {
-        document.body.classList.remove('app__body', 'channel-view');
+    // OS Detection
+    if (platform === 'Win32' || platform === 'Win64') {
+        bodyClass.push('os--windows');
+    } else if (platform === 'MacIntel' || platform === 'MacPPC') {
+        bodyClass.push('os--mac');
     }
 
-    render() {
-        const shouldShowAppBar = this.props.shouldShowAppBar;
-
-        return (
-            <div
-                id='channel_view'
-                className='channel-view'
-            >
-                <AnnouncementBarController/>
-                <SystemNotice/>
-                <FaviconTitleHandler/>
-                <ProductNoticesModal/>
-                <div className={classNames('container-fluid channel-view-inner', {'app-bar-enabled': shouldShowAppBar})}>
-                    <SidebarRight/>
-                    <SidebarRightMenu/>
-                    <Sidebar/>
-                    {!this.props.fetchingChannels && <Route component={CenterChannel}/>}
```

Comment:
All components mounted under a Router will have access to match, params etc when used with withRouter and router hooks inside of component. We dont have to pass it under Route component.